### PR TITLE
experiment(http): tower compression

### LINKED ISSUE #13835: Streaming HTTP response body compression
Fixed length response bodies were done in https://github.com/denoland/deno/pull/13769

### PR DESCRIPTION
**Pull request by AaronO**
Simpler, more composable & adds stream support "for free", fixes #13835

Invariants we may want to test are preserved (across main and alt impls, i.e: #14325, [tree/compression_stream](https://github.com/denoland/deno/tree/compression_stream)):
- http performance / throughput
- etag weakening ? (not done atm in this PR, but could be)
- compression predicates (~same responses are compressed)

### CODE REVIEW

File: ext/http/lib.rs [hunk 1]
```diff
@@ -101,6 +102,36 @@ impl From<tokio::net::unix::SocketAddr> for HttpSocketAddr {
   }
 }
 
+// implements tower_http::compression's Predicate
+fn not_uncompressible(
```
Comment (lucacasonato) [inline ext/http/lib.rs:106]: not_uncompressible == is_compressible? double negation is very weird to read IMO

**Review by lucacasonato — COMMENTED** [review]
Does this handle `vary` and `etag` headers like we do now?

File: ext/http/lib.rs (hunk 1 above)
Comment (AaronO) [reply ext/http/lib.rs]: Yeah agreed, had fixed this yesterday but hadn't pushed

**Comment by bartlomieju** [comment]
Do we still want to land this PR, given that #14345 landed?

**Pull request closed without merging**